compiler/rustc_passes/src/stability.rs

```diff
@@ -300,11 +300,27 @@ impl<'a, 'tcx> Annotator<'a, 'tcx> {
         // const stability: inherit feature gate from regular stability.
         let mut const_stab = const_stab.map(|(stab, _span)| stab);
 
+        // `impl const Trait for Type` items forward their const stability to their
+        // immediate children.
+        // FIXME(const_trait_impl): how is this supposed to interact with `#[rustc_const_stable_indirect]`?
+        // Currently, once that is set, we do not inherit anything from the parent any more.
+        if const_stab.is_none()
+            && let Some(parent) = self.parent_const_stab
+            && parent.is_const_unstable()
+        {
+            // For now, `const fn` in const traits/trait impls does not exist.
+            assert!(
+                fn_sig.is_none_or(|s| !s.header.is_const()),
+                "should never have parent const stability for a const fn"
+            );
+            self.index.const_stab_map.insert(def_id, parent);
```

Comment:
Shouldn't this be `const_stab = parent`?

Right now, it seems like we could then enter the `if` below which actually sets `const_stab` to `Some` and then *that* gets inserted into the table.